src/System.Console/src/System/Console.cs

```diff
@@ -38,10 +43,65 @@ public static TextReader In
         {
             get
             {
-                return Volatile.Read(ref _in) ?? EnsureInitialized(ref _in, () =>
+                return Volatile.Read(ref s_in) ?? EnsureInitialized(ref s_in, () => ConsolePal.GetOrCreateReader());
```

Comment:
That was my initial plan as I thought it would be more intuitive and easy to track. However, in our design since we pass these across for example to generic method, anonymous delegate, I found explicit calls much more easy to validate in my head as I read the code, so I chose that instead and made sure we have the right barriers wherever required :)